Add: Include basic .editorconfig file to help enforce tab style

orudge opened the pull request:
As per http://editorconfig.org/, add a .editorconfig file to help coax editors into doing the right thing by default. :)

orudge committed:
Add: Include basic .editorconfig file to help enforce tab style

TrueBrain commented on .editorconfig:
```diff
@@ -0,0 +1,8 @@
+root = true
+
+[*]
+insert_final_newline = true
```
is eol also a good idea here?

glx22 replied:
the doc says to not set eol if we want native line endings

TrueBrain commented on .editorconfig:
```diff
@@ -0,0 +1,8 @@
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.{c,cpp,h,hpp}]
```
how about charsets for these?

orudge replied:
I can add charset = utf-8 :)

orudge committed:
Add: Require UTF-8 for code files in .editorconfig

frosch123 commented:
Wouldn't the charset go into the [*] section?
All our files should be UTF-8.

orudge commented:
@frosch123 Generally, I would say so, yes. I didn't want to be too prescriptive in this file though in case, for any reason, we had some files that it caused problems for. (Windows .rc files, for instance, don't like UTF-8). There are no doubt other files or file types that could be added; this is just intended to provide an initial foundation.

TrueBrain reviewed (dismissed):
You give a finger .. and we want to eat the hand :D

TrueBrain commented:
Sorry, there was this "update branch" button I just had to click :D Please rebase your branch so you can merge it :)

orudge commented:
I've removed your merge and rebased this onto the current OpenTTD master, is that what you meant? Also the CI checks have failed so the merge option is greyed out, unless you just want me to push directly to the repository?
